semeio/workflows/ahm_analysis/ahmanalysis.py

```diff
@@ -0,0 +1,456 @@
+import collections
+import tempfile
+import glob
+import os
+import itertools
+import numpy as np
+import pandas as pd
+import warnings
+
+from pathlib import Path
+from scipy.stats.stats import ks_2samp
+from sklearn.decomposition import PCA
+from sklearn.preprocessing import StandardScaler
+
+from xtgeo.grid3d import Grid
+from xtgeo.grid3d import GridProperty
+from semeio.communication import SemeioScript
+from semeio._exceptions.exceptions import ValidationError
+from res.enkf import (
+    ErtImplType,
+    EnkfNode,
+)
+from res.enkf import (
+    ESUpdate,
+    ErtRunContext,
+)
+from res.enkf.export import (
+    GenKwCollector,
+    MisfitCollector,
+)
+
+from ert_shared.plugins.plugin_manager import hook_implementation
+from ert_shared.libres_facade import LibresFacade
+
+
+DESCRIPTION = """
+Purpose is to calculate (ERT workflow)
+the degree of update (KS matrix)
+of the different parameters with regards to the different
+observations. Create csv files which can be used in
+Webviz/Jupyternotebook/dash to perform analysis of the AHM.
+requires an ert config file as input
+optionally a name can be set for the target case used for
+update (ahm) calculation
+"""
+
+
+class AhmAnalysisJob(SemeioScript):  # pylint: disable=too-few-public-methods
+    """ Define ERT workflow to evaluate change of parameters for each observation"""
+
+    def run(self, target_name="analysis_case", prior_name=None, group_by="data_key"):
+        """Perform analysis of parameters change per obs group
+        prior to posterior of ahm"""
+        ert = self.ert()
+        facade = LibresFacade(self.ert())
+
+        obs_keys = [
+            facade.get_observation_key(nr)
+            for nr, _ in enumerate(facade.get_observations())
+        ]
+        key_map = _group_observations(facade, obs_keys, group_by)
+
+        prior_name, target_name = check_names(
+            facade.get_current_case_name(),
+            prior_name,
+            target_name,
+        )
+        # Get the prior scalar parameter distributions
+        prior_data = check_inputs(
+            MisfitCollector.loadAllMisfitData(ert, prior_name),
+            GenKwCollector.loadAllGenKwData(ert, prior_name),
+        )
+
+        # create dataframe with observations vectors (1 by 1 obs and also all_obs)
+        combinations = make_obs_groups(key_map)
+        # setup dataframe for calculated data
+        ks_data, active_obs, misfitval = initialize_emptydf()
```

Comment:
I think we can replace this with e.g. `ks_data, active_obs, misfitval = pd.DataFrame(), pd.DataFrame(), pd.DataFrame()`

It's fairly trivial, and we don't need specific tests for that function.

Reply:
done